srcmedia/ts/lib/form.ts

```diff
@@ -0,0 +1,59 @@
+import { Subject } from 'rxjs'
+
+import { Reactive, Component } from './common'
+
+interface RxFormState {
+    // eventually this could include generic things like validity
+}
+
+class RxForm extends Component implements Reactive<RxFormState> {
+    element: HTMLFormElement
+    state: Subject<RxFormState>
+    target: string
+
+    constructor(element: HTMLFormElement) {
+        super(element)
+        this.state = new Subject()
+        // FIXME naively assuming that the form targets the current path
+        this.target = window.location.pathname
+    }
+    /**
+     * Update the form's state. We don't need to change the DOM, so all we do
+     * is call next().
+     *
+     * @param {RxFormState} state
+     * @returns {Promise<void>}
+     * @memberof RxForm
+     */
+    async update(state: RxFormState): Promise<void> {
+        this.state.next(state)
```

Comment:
Should I know what calling `next()` does? Is there context I'm missing to understand this?

Reply:
it's a method on all Observable-like datatypes (e.g. `Subject`) that pushes the next item into the sequence, analogous to `self.next()` on a python generator.